engine/src/flutter/lib/web_ui/lib/src/engine/arena.dart

```diff
@@ -0,0 +1,22 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+abstract class Collectable {
+  void collect();
+}
+
+class Arena {
+  List<Collectable> _collectables = <Collectable>[];
```

Comment:
could be fine 😄

Reply:
No, it can't be final. On collection, we replace it with an empty list before collecting all the objects. This is just to make this class safe against re-entrancy, so that if an object is added to the arena in the collect method of one of the collectables, it adds to the new empty list rather than the list we're iterating (we don't want to mutate while iterating).

Reply:
Oh! carry on.